web/ui/react-app/src/pages/graph/Panel.tsx

```diff
@@ -138,55 +144,72 @@ class Panel extends Component<PanelProps, PanelState> {
         throw new Error('Invalid panel type "' + this.props.options.type + '"');
     }
 
-    fetch(`${this.props.pathPrefix}/${API_PATH}/${path}?${params}`, {
-      cache: 'no-store',
-      credentials: 'same-origin',
-      signal: abortController.signal,
-    })
-      .then(resp => resp.json())
-      .then(json => {
-        if (json.status !== 'success') {
-          throw new Error(json.error || 'invalid response JSON');
-        }
-
-        let resultSeries = 0;
-        if (json.data) {
-          const { resultType, result } = json.data;
-          if (resultType === 'scalar') {
-            resultSeries = 1;
-          } else if (result && result.length > 0) {
-            resultSeries = result.length;
-          }
+    let query;
+    let exemplars;
+    try {
+      query = await fetch(`${this.props.pathPrefix}/${API_PATH}/${path}?${params}`, {
+        cache: 'no-store',
+        credentials: 'same-origin',
+        signal: abortController.signal,
+      }).then(resp => resp.json());
+
+      if (query.status !== 'success') {
+        throw new Error(query.error || 'invalid response JSON');
+      }
+
+      if (showExemplars) {
```

Comment:
I get the intent to group everything graph-related under the switch case above, but at the point where we are actually deciding whether to run a query at all or not, I wonder if it wouldn't be more straightforward to understand if we just omitted the extra variable and just did this here:

```suggestion
      if (this.props.options.type === 'graph' && this.props.showExemplars) {
```